### PR TITLE
TDL-18856: Fix record write for parent of grand child stream.

### PR DESCRIPTION
**Pull request by savan-chovatiya**
# Description of change
Resolution:

- Updated write_schema and sync function to write records and schema only if any stream is selected in the catalog(parent stream of the grandchild).

The scenario covered in manual testing:
- Verified that if grandchild stream is selected then write_schema is called for only it and not for parent stream of it

# Manual QA steps
 - 
 
# Risks
 - 
 
# Rollback steps
 - revert this branch

### CODE REVIEW

**Review by hpatel41 — APPROVED** [review]

**Review by KrisPersonal — APPROVED** [review]

**Review by kspeer825 — CHANGES_REQUESTED** [review]
Independent selection of parent and child streams can be covered by tap-tester. Could you add a test that selects the child stream and not the parent stream and confirm that the sync runs successfully.

**Review by RushiT0122 — APPROVED** [review]

**Comment by savan-chovatiya** [comment]
> Independent selection of parent and child streams can be covered by tap-tester. Could you add a test that selects the child stream and not the parent stream and confirm that the sync runs successfully.

@kspeer825 I added an integration test to confirm that the sync runs fine if only child streams(all levels) are selected.

File: .circleci/config.yml [hunk 1]
```diff
@@ -22,20 +22,25 @@ jobs:
...
+          command: |
+            source /usr/local/share/virtualenvs/tap-snapchat-ads/bin/activate
+            pip install coverage
+            nosetests --with-coverage --cover-erase --cover-package=tap_snapchat_ads --cover-html-dir=htmlcov tests/unittests
+            coverage html
+      - store_test_results:
+          path: test_output/report.xml
+      - store_artifacts:
+          path: htmlcov
       - add_ssh_keys
```
Comment (kspeer825) [inline .circleci/config.yml:36]: The `add_ssh_keys` is not necessary

File: tests/test_parent_child_independent.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        second_level_child_streams = {"ad_account_stats_daily", "ad_account_stats_hourly", "audience_segments", "pixels", "media",
+                               "creatives", "phone_numbers", "campaigns", "ad_squads", "ads", "product_sets"}
+        # Don't have data for stats stream so excluding from test
+        self.run_test(second_level_child_streams - self.stats_streams)
+        
+        # select third_level_child_streams only and run test
+        third_level_child_streams = {"pixel_domain_stats", "campaign_stats_daily", "campaign_stats_hourly", "ad_squad_stats_daily",
+                                     "ad_squad_stats_hourly", "ad_stats_daily", "ad_squad_stats_hourly"}
+        # Don't have data for stats streams so excluding from test(all streams are stats stream for third_level_child_streams)
+        # self.run_test(third_level_child_streams - self.stats_streams)
```
Comment (kspeer825) [inline tests/test_parent_child_independent.py:24]: Could you adjust this pattern to create a new `test_` method for each case that then calls the `run_test`.

**Review by kspeer825 — CHANGES_REQUESTED** [review]
Couple of minor suggestions.

File: .circleci/config.yml (hunk 1 above)
Comment (savan-chovatiya) [reply .circleci/config.yml]: Removed add_ssh_keys from config.yml.

File: tests/test_parent_child_independent.py (hunk 2 above)
Comment (savan-chovatiya) [reply tests/test_parent_child_independent.py]: Updated pattern as suggested by adding test_ method for all three scenarios.

**Review by kspeer825 — APPROVED** [review]
Awesome!

**Pull request merged**